appinfo/app.php

```diff
@@ -26,9 +26,28 @@
 if(OC::$CLI) {
 	return;
 }
-
 $config = \OC::$server->getConfig();
 $gssMode = $config->getSystemValue('gss.mode', '');
 if ($gssMode === 'master') {
 	return;
 }
+
+$userSession = \OC::$server->getUserSession();
+$masterUrl = $config->getSystemValue('gss.master.url', '');
+$request = \OC::$server->getRequest();
+if (!$userSession->isLoggedIn() && !empty($masterUrl) &&
+	$request->getPathInfo() === '/login') {
+
+	// an admin wants to login directly at the Nextcloud node
+	$params = $request->getParams();
+	if (isset($params['direct'])) {
+		return;
+	}
+
+	if(isset($params['redirect_url'])) {
+		$masterUrl .= $params['redirect_url'];
```

Comment:
Ummm ... I guess this is not how it works, right? It needs to be `?redirect_url=URL` or am I wrong?

Reply:
yes that makes sense...

Reply:
@schiessle ping?

Reply:
hm, I'm confused... Do you think it works or not... I remember that I tested it. But if you elaborate your exact concerns I can try it again.

Reply:
Assume

```php
$masterUrl = 'https://master.gss'
$params['redirect_url'] = 'foo/bar';
```

Then with this code it becomes `https://master.gssfoo/bar`
It should be `https://master.gss?redirect_url=foo/bar`

Reply:
@rullzer no, I tested it. Look for example how internal file links look like `https://yourcloud.com/f/843046` and not `https://yourcloud.com?redirect_url=/f/843046`. If you use the second one we will strip the redirect_url away and you will arrive at this URL `https://yourcloud.com/login`.

If you use `https://yourcloud.com/f/843046` you will arrive at `https://yourcloud.com/login?redirect_url=/f/843046` and we will process it correctly in the login process, redirect to the GSS node, etc.

Reply:
Ah right. Of course there is other magic at play here. Yes what you say makes sense.

Reply:
Ah - so there is another round trip because you are not logged in and then the page needs auth and creates the redirect URL. Makes sense 👍